src/components/DocumentationSettings/DocumentationActions.tsx

```diff
@@ -0,0 +1,117 @@
+import {
+  Button,
+  Dialog,
+  DropdownMenu,
+  Flex,
+  Text,
+  TextField,
+} from "@radix-ui/themes";
+import { DocumentationSource } from "./DocumentationSettings";
+import { useState } from "react";
+
+type DocumentationActionsProps = {
+  source: DocumentationSource;
+  deleteDocumentation: (url: string) => void;
+  editDocumentation: (url: string, maxDepth: number, maxPages: number) => void;
+  refetchDocumentation: (url: string) => void;
+};
+
+export const DocumentationActions: React.FC<DocumentationActionsProps> = ({
+  source,
+  deleteDocumentation,
+  editDocumentation,
+  refetchDocumentation,
+}: DocumentationActionsProps) => {
+  const [maxDepth, setMaxDepth] = useState(source.maxDepth);
+  const [maxPages, setMaxPages] = useState(source.maxPages);
+  const [isDialogOpen, setIsDialogOpen] = useState(false);
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+
+  return (
+    <>
+      <DropdownMenu.Root onOpenChange={setIsDropdownOpen}>
+        <DropdownMenu.Trigger>
+          <Button variant="soft">
+            Actions
+            <DropdownMenu.TriggerIcon />
+          </Button>
+        </DropdownMenu.Trigger>
+        <DropdownMenu.Content>
+          <DropdownMenu.Item
+            onSelect={() => {
+              setIsDialogOpen(true);
+            }}
+          >
+            Edit
+          </DropdownMenu.Item>
+          <DropdownMenu.Item onSelect={() => refetchDocumentation(source.url)}>
+            Refetch
+          </DropdownMenu.Item>
+          <DropdownMenu.Separator />
+          <DropdownMenu.Item
+            color="red"
+            onClick={() => deleteDocumentation(source.url)}
+          >
+            Delete
+          </DropdownMenu.Item>
+        </DropdownMenu.Content>
+      </DropdownMenu.Root>
+      <Dialog.Root
+        open={isDialogOpen && !isDropdownOpen}
+        onOpenChange={setIsDialogOpen}
+      >
+        <Dialog.Content maxWidth="450px">
+          <Dialog.Title>{`Edit ${source.url}`}</Dialog.Title>
+          <Flex direction="column" gap="3">
+            <label>
```

Comment:
Text can be passed `as="label"`

```suggestion
  <Text as="label" size="2" mb="1" weight="bold">
     Max depth
     <TextField.Root
          defaultValue={maxDepth}
           onChange={(change) => setMaxDepth(Number(change.target.value))}
           type="number"
            placeholder="Enter the max depth"
      />
</Text>
```